### PR TITLE
Add general Xamarin sample for vision models.

### PR DESCRIPTION
**Pull request by skottmckay**
Currently demonstrates image classification and face recognition.
More models can be added in the future for SSD etc.
Setup to be able to get ORT from the INT NuGet repo for pre-release testing, and the main NuGet repo for official release testing.

### CODE REVIEW

**Comment by gwang-msft** [comment]
Small nit. can we add a sample image for UltraFace, in case both pick and capture fail?
Something like these sample image here? https://github.com/onnx/models/tree/f064171f7dd8e962a8a5b34eac8e1bcf83cebbde/vision/body_analysis/ultraface/dependencies

**Review by gwang-msft — APPROVED** [review]
LGTM

**Comment by gwang-msft** [comment]
Also, we may need to add license header for most of the code files

File: mobile/examples/Xamarin/readme.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+## Getting Started
+
+> [!IMPORTANT] 
+> The nuget.config has package sources for the official and integration NuGet repositories. 
+> Xamarin support is currently only available in the ONNX Runtime nuget packages in the integration repository. It will be added to the official package in the v1.10 release.
+>
+> There are some [known issues](#known-issues) that could impact aspects of the sample on specific devices or environments. See [known issues section](#known-issues) for workarounds.
+
+The [VisionSample](VisionSample/VisionSample.sln) should build and run as-is, and looks for model files in a folder in this directory called **Models**. 
+The Ultraface model should already exist in that directory and to able to be used.
```
Comment (YUNQIUGUO) [inline mobile/examples/Xamarin/readme.md:28]: Not sure why include the Ultraface model in the directory but the ResNet need to be downloaded? Thought might be nice to do it in a consistent way.

Comment (skottmckay) [reply mobile/examples/Xamarin/readme.md:28]: Ultra face is small. ResNet is nearly 100MB

**Review by YUNQIUGUO — APPROVED** [review]
LGTM too.

File: mobile/examples/Xamarin/VisionSample/VisionSample/VisionSampleBase.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                                if (executionProvider == ExecutionProviderOptions.CPU)
+                                {
+                                    // create session that uses the CPU execution provider
+                                    _session = new InferenceSession(_model);
+                                }
+                                else
+                                {
+                                    // create session that uses the NNAPI/CoreML. the CPU execution provider is also
+                                    // enabled by default to handle any parts of the model taht NNAPI/CoreML cannot.
+                                    var options = SessionOptionsContainer.Create(nameof(ExecutionProviderOptions.Platform));
```
Comment (YUNQIUGUO) [inline mobile/examples/Xamarin/VisionSample/VisionSample/VisionSampleBase.cs:52]: nit: taht -> that

File: mobile/examples/Xamarin/VisionSample/VisionSample/SkiaSharpImageProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+using Microsoft.ML.OnnxRuntime.Tensors;
+using SkiaSharp;
+
+namespace VisionSample
+{
+    public class SkiaSharpImageProcessor<TPrediction, TTensor> : IImageProcessor<SKBitmap, TPrediction, TTensor>
+    {
+        protected virtual SKBitmap OnPreprocessSourceImage(SKBitmap sourceImage) => sourceImage;
+        protected virtual Tensor<TTensor> OnGetTensorForImage(SKBitmap image) => throw new NotImplementedException();
+        protected virtual void OnPrepareToApplyPredications(SKBitmap image, SKCanvas canvas) { }
```
Comment (YUNQIUGUO) [inline mobile/examples/Xamarin/VisionSample/VisionSample/SkiaSharpImageProcessor.cs:13]: Assume this should be `OnPrepareToApplyPredictions` ?

predications -> predictions?

**Pull request merged**